satpy/resample.py

```diff
@@ -1046,6 +1050,23 @@ def compute(self, data, expand=True, **kwargs):
         return update_resampled_coords(data, new_data, target_geo_def)
 
 
+def _get_arg_to_pass_for_skipna_handling(skipna, **kwargs):
+    """Determine if skipna can be passed to the compute functions for the average and sum bucket resampler."""
+    # FIXME this can be removed once Pyresample 1.18.0 is a Satpy requirement
+    if 'mask_all_nan' in kwargs:
+        if PR_USE_SKIPNA:
+            warnings.warn('Argument mask_all_nan is deprecated. Please use skipna for missing values handling. '
+                          'Continuing with default skipna=True, if not provided differently.', DeprecationWarning)
```

Comment:
You could add a line here for `skipna = kwargs['mask_all_nan']` followed by `return `{'skipna': skipna}`, right?

Then in the `else` below this do `return kwargs`...I think. Basically I think you can combine these if statements.

Additionally, I think you need to update the `kwargs` dictionary instead of only returning a specific dictionary. Otherwise you lose any additional arguments, right?

Reply:
Maybe the arguments to this function should just be `**kwargs`?

Reply:
Thanks for the review!
I rearranged a bit the logic and now I'm handling the kwargs only, I think it makes sense too.
I also rearranged the if statements a bit and avoided one `else` now. I think this cannot be reduced further, if we want to issue the warnings only if the `mask_all_nan` was passed by the user. I also added tests for this logic.